### DISCUSSION TITLE
Actions does not run if update is overridden?

### DISCUSSION THREAD

**Discussion by nilskj** [excaliburjs/Excalibur#2059] [Q&A] [answered]
Hey! Not sure if bug or not ;) I had some problems getting my Actions to run. When troubleshooting I found out that I had override `update` for some basic inputs:
```javascript
  update(engine: Engine, delta: number) {
    if (engine.input.keyboard.isHeld(Input.Keys.W)) {
      this.velocity = Vector.Up;
    } else if (engine.input.keyboard.isHeld(Input.Keys.A)) {
      this.velocity = Vector.Left;
    } else if (engine.input.keyboard.isHeld(Input.Keys.D)) {
      this.velocity = Vector.Right;
    } else if (engine.input.keyboard.isHeld(Input.Keys.S)) {
      this.velocity = Vector.Down;
    }
  }
```

I were able to create and queue actions but no actions fired... There is a system to run actions in update that is abstracted away but I had no clue this was causing it. To make actions run I needed to call the super update as well, is this intentional? If so this could be explained in Actions docs section to prevent this pitfall.

```javascript
  update(engine: Engine, delta: number) {
    ...inputs
    super.update(engine,delta);
  }
```

**Comment by eonarheim** [comment, marked as answer]
Hi @nilskj apologies for the pitfall, this is indeed the case in the current version of Excalibur that the `super.update()` must be called to preserve the action behavior.

Actions is one of the last things to be refactored out of the `Actor.update()`, the plan is to implement the Actions API with an ECS Component/System in the next version removing the need to call the `super.update()`.

I've added a couple updates to the documentation to note this requirement

* https://excaliburjs.com/docs/actors#update
* https://excaliburjs.com/docs/actions#actions
